Add support for valgrind outputs filtered by CTest/CDash

LucHermitte opened the pull request:
This is a first PR.

 I'll eventually have other(s?) that move the functions to autoload plugins, and that provide options to chose the keybindings (a mapping on `<space>` with this new trend to set the leader to `<space>` can be problematic).

LucHermitte committed:
Bang functions in order to be able to reload the plugin

LucHermitte committed:
Ignore CTest prefixes (by trimming them)

CTest prefix its outputs with `\d\+:`. As we may work on valgrind files
produced though CTest/CDash, we need to fix the file before doing anything with
them.

LAHumphreys reviewed (approved):
Thanks, I've merged the function bangs in.

LAHumphreys commented on plugin/valgrind.vim:
```diff
@@ -123,8 +123,8 @@ function! s:Valgrind( ... )
     endif
-    silent execute 'g!/^==\d*==/d'
-    silent execute '%s/^==\d*== //e'
+    silent execute 'g!/^\(\d\+: \)\===\d*==/d'
```
I agree the use case is valid, but I'm not sure hard-coding the optional prefix here is the right answer - would we be better off having a prefix variable which can be configured in a vimrc?

LucHermitte replied:
Either that or a hook that removes noise. For instance, I'm thinking of a possible encapsulation of each line by an XML node. In that case, we'll need to pre-process the lines differently.

LucHermitte committed:
Add a hook to restore valgrind output

Tools like CTest alter valgrind output. Other tools may encapsulate each line
differently -- for instance, they could be wrapped in XML nodes.

Hence this new option.

LucHermitte committed:
Update author list

Grauniad commented:
Nice - thanks, I've pulled this in :)

I agree with you that making these key binding configurable would be valuable.
